### PR TITLE
Avoid using std::cout in a test.

### PR DESCRIPTION
**Pull request by drwells**
This is problematic on misconfigured MPI implementations that print warnings for one reason or another every time they are run (e.g., the default MPI implementation on Ubuntu 16.04).

It is easy enough in this case to do the normal thing and print everything to deallog.

I think that the small number of failing MPI tests on this platform all fail for this reason.

### CODE REVIEW

File: tests/mpi/petsc_step-27.cc [hunk 1]
```diff
@@ -329,24 +329,24 @@ namespace Step27
...
     preconditioner.initialize(system_matrix, data);
 
-    check_solver_within_range(cg.solve(system_matrix,
-                                       completely_distributed_solution,
-                                       system_rhs,
-                                       preconditioner),
-                              solver_control.last_step(),
-                              5,
-                              40);
+    cg.solve(system_matrix,
```
Comment (tjhei) [inline tests/mpi/petsc_step-27.cc:346]: why this change?

Comment (drwells) [reply tests/mpi/petsc_step-27.cc:346]: Since the test copies stdout/stderr to `output` (i.e., things printed to deallog were ignored) that check never made it to the stored `.output` files. I deleted it so that the test had the same output (aside from blank lines) as it did before.
